### DISCUSSION TITLE
Setting My Container's Height Makes Some of Them Invisible

### DISCUSSION THREAD

**Discussion by eeriemyxi** [Textualize/textual#4108] [General]
https://github.com/Textualize/textual/assets/86738916/224cbb92-d746-4401-bef9-14d20d09d81f

I never tried front-end stuff before, maybe it's the norm there, but I honestly have no idea why this is happening. I am trying to set a fix height for each task, but when I do, some of them disappear for whatever reason.

This is how my task containers are generated:
```python
def make_todo_task_container(db, code, name):
    container = HorizontalScroll(classes="todo_task_container", id=code)

    container.mount_all(
        [
            Label(code, classes="todo_task_code_label"),
            Label(name, classes="todo_task_name_label"),
        ]
    )

    add_date_label_to_container(db, code, container, 1)
    add_repeat_label_to_container(db, code, container, 1 + (db[code]["notify_after"] == -1))

    return container
```
And this is the container that keeps the task containers:
```python
def make_todo_container(todo_list):
    todo_container = VerticalScroll(id="todo_container", classes="todo_container")
    for code, info in todo_list.items():
        task_container = make_todo_task_container(todo_list, code, info["title"])
        task_container.visible = info["visible"]
        todo_container.mount(task_container)
    return todo_container
```
This is how I am setting the height:
```css
    .todo_task_container {
        height: 5%;
    }
```

I don't know what else I should be sharing, please let me know. Also, sorry about the crappy video quality, I did try messing with the capture quality settings, but it didn't improve much.

**Comment by davep** [comment]
I've watched the video but I can't follow what you're trying to do, what you consider working, what you consider not working, what the problem is or what you're trying to actually achieve. Do you think you could perhaps try and [make an MRE](https://label.dev/articles/minimal-reproducible-example/) of what you're trying to do and clearly document what isn't working?

As an aside, you may also find [the howto on designing a layout](https://textual.textualize.io/how-to/design-a-layout/) a helpful read.

**Reply by eeriemyxi** [reply]
I managed to fix it. Thanks. The issue was that I was using percentages to set the fixed height of my containers; then I used explicit integers, and it worked.
![image](https://github.com/Textualize/textual/assets/86738916/48e107b6-84af-40e9-85d5-5cab00fafdbd)

**Reply by davep** [reply]
Awesome; glad you got to the bottom of it.

This is looking really nice by the way! I love the colour scheme.
